### DISCUSSION TITLE
Edit Tailwind Classes After Deployment For Production

### DISCUSSION THREAD

**Discussion by alibabaeefml** [tailwindlabs/tailwindcss#10680] [Help] [answered]
I was wondering how to edit styles by classes processed by tailwind cli after build for production? 
I mean I'd rather not have to use "npx tailwind ...." on everyone's device to update the output.css in dist folder so my new styles would apply.

I came with using tailwind.js from cdn but tailwind says it's not ok for production. And I wonder what problems I may face using this approach? Also what's the alternative?

**Comment by wongjn** [comment, marked as answer]
> I was wondering how to edit styles by classes processed by tailwind cli after build for production?

You could have an additional build step after the `tailwind build` command. This step would take the output file from `tailwind build` and process it further as you'd like.

> I mean I'd rather not have to use "npx tailwind ...." on everyone's device to update the output.css in dist folder so my new styles would apply.

I'm not sure what you mean by <q>everyone's device</q> here. You should only need to run `npx tailwind …` in production and the changes should be reflected (as long as correct `Cache-Control` headers and/or cache-busting techniques are used).

> I came with using tailwind.js from cdn but tailwind says it's not ok for production. And I wonder what problems I may face using this approach?

You see the webpage unstyled until JavaScript from the CDN has been executed. I would expect this to be judged as not aesthetically pleasing. Having the compiled CSS instead means that the CSS can be downloaded and applied to the webpage without the need for any JavaScript execution.

> Also what's the alternative?

Use some sort of build system that can integrate PostCSS or use the Tailwind CLI to build CSS.

**Reply by alibabaeefml** [reply]
Thanks for your time.

So I'm gonna install tailwind cli on my website server and run the npx watcher command so anytime I add tailwind classes it compiles them automatically. Is that OK?

**Reply by wongjn** [reply]
I wouldn't say so. You'd have the Tailwind CLI locally on your development machine and run the watcher while you're in development. Then once the changes are complete, run a build and upload the built CSS to the website server.
